src/main/java/org/opensearch/ad/transport/SearchTopAnomalyResultRequest.java

```diff
@@ -170,8 +171,24 @@ public static SearchTopAnomalyResultRequest parse(XContentParser parser, String
         return new SearchTopAnomalyResultRequest(detectorId, taskId, historical, size, convertedCategoryFields, order, startTime, endTime);
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeOptionalString(detectorId);
+        out.writeOptionalString(taskId);
+        out.writeBoolean(historical);
+        out.writeOptionalInt(size);
+        out.writeOptionalStringArray((String[]) categoryFields.toArray());
```

Comment:
Have you tested the (de)serialization functionality? You write as a string array and read it as a list, would that work?